vllm/model_executor/layers/fused_moe/fused_moe.py

```diff
@@ -377,34 +377,39 @@ def fused_topk(
     gating_output: torch.Tensor,
     topk: int,
     renormalize: bool,
+    routing_func: Callable = torch.topk,
 ):
     assert hidden_states.shape[0] == gating_output.shape[0], (
         "Number of tokens mismatch")
 
     M, _ = hidden_states.shape
 
-    topk_weights = torch.empty(M,
+    if routing_func != torch.topk:
```

Comment:
This is slightly confusing. Is there a reason this check can't be done in `select_experts` where the routing function can then be called, instead of passing a callable into another callable? I think we should fused_topk as is.